modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/MessageBoardMessageDTOConverter.java

```diff
@@ -153,6 +160,24 @@ public MessageBoardMessage toDTO(DTOConverterContext dtoConverterContext)
 		};
 	}
 
+	private String _getCompanyMxName(long companyId, long groupId, User user)
+		throws Exception {
+
+		MBModerationGroupConfiguration mbModerationGroupConfiguration =
+			_configurationProvider.getGroupConfiguration(
+				MBModerationGroupConfiguration.class, groupId);
+
+		Company company = _companyLocalService.getCompany(companyId);
+
+		if (company.hasCompanyMx(user.getEmailAddress()) &&
+			mbModerationGroupConfiguration.enableCompanyMx()) {
```

Comment:
I my humble opinion, we should check first is this configuration is enabled, because it's a key check for the flow, it may "saves" unnecessary processing

What you guys think? @nNilton , @gleicelisbino  

```suggestion
		if (mbModerationGroupConfiguration.enableCompanyMx()) &&
			company.hasCompanyMx(user.getEmailAddress()) {
```